libs/datamodel/parser-database/src/walkers/composite_type.rs

```diff
@@ -118,6 +118,19 @@ impl<'db> CompositeTypeFieldWalker<'db> {
         &self.field.r#type
     }
 
+    /// The type of the field in case it is a scalar type (not an enum, not a composite type).
+    pub fn scalar_type(self) -> Option<ScalarType> {
+        let mut r#type = self.r#type();
+
+        loop {
+            match r#type {
+                ScalarFieldType::BuiltInScalar(scalar) => return Some(*scalar),
+                ScalarFieldType::Alias(alias_id) => r#type = &self.db.types.type_aliases[alias_id],
```

Comment:
We already catch loops somewhere else?

Reply:
This one just resolves the actual type. Copied from the `scalar_type.rs`.

Reply:
Loops are catched way earlier.